File: Digito Verificador/Modulo 10/Teste - Arquivo 10/GerarListaMatriculasMod10.go
package main

import (
	"fmt"
	"os"
)

func formataMatricula(MATRICULA int) string {
	return fmt.Sprintf("%06d", MATRICULA)
}

func mod10(MATRICULA string) int {
	SOMA := 0
	for I := 0; I <= 5; I++ {
		DIGITO := int(MATRICULA[I] - '0')
		if I % 2 != 0 && DIGITO * 2 > 9 {
			SOMA += DIGITO * 2 - 9
		} else if I % 2 != 0 && DIGITO * 2 <= 9 {
			SOMA += DIGITO * 2
		} else {
			SOMA += DIGITO
		}
	}
	DV := (SOMA % 10)
	if DV != 0 {
		DV = 10 - DV
	}
	return DV
}

func main() {
	ARQUIVO, ERRO := os.Create("numeros.txt")
	if ERRO != nil {
		fmt.Println("Erro ao abrir o ARQUIVO.")
		fmt.Println("Programa encerrado.")
		return
	}
	defer ARQUIVO.Close()

	fmt.Println("GERADOR DE NUMEROS DE MATRICULA DE 000.000 ATE 999.999 COM \"DV\"")
	fmt.Println("Padrao: Modulo 10")
	fmt.Println("---------------------------------------------------------------")
	fmt.Println()

	fmt.Print("Aguarde ARQUIVO sendo gerado... ")

	for I := 0; I <= 999999; I++ {
		MATRICULA := formataMatricula(I)
		DV := mod10(MATRICULA)
		fmt.Fprintf(ARQUIVO, "%s.%s-%d\n", MATRICULA[:3], MATRICULA[3:], DV)
	}

	fmt.Println("\n\nSUCESSO: Arquivo gerado.\n")
}

File: Digito Verificador/Modulo 10/Teste - Arquivo 10/GerarListaMatriculasMod10.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char* formataMatricula(int MATRICULA) {
  char* str = (char*)malloc(7 * sizeof(char));
  snprintf(str, 7, "%06d", MATRICULA);
  return str;
}

int mod10(const char* MATRICULA) {
  int SOMA = 0;
  for (int I = 0; I < 6; I++) {
    int DIGITO = MATRICULA[I] - '0';
    if (I % 2 != 0 && DIGITO * 2 > 9)
      SOMA += DIGITO * 2 - 9;
    else if (I % 2 != 0 && DIGITO * 2 <= 9)
      SOMA += DIGITO * 2;
    else
      SOMA += DIGITO;
  }
  int DV = (SOMA % 10 == 0) ? 0 : 10 - SOMA % 10;
  return DV;
}

int main(void) {
	
  char MATRICULA_TEMP[10];
  int I;	
  FILE* ARQUIVO = fopen("numeros.txt", "w");

  printf("GERADOR DE NUMEROS DE MATRICULA DE 000.000 ATE 999.999 COM \"DV\"\n");
  printf("Padrao: Modulo 10\n");
  printf("---------------------------------------------------------------\n\n");
  
  if (ARQUIVO == NULL) {
    fprintf(stderr, "Erro ao abrir o arquivo.\n");
    printf("Programa encerrado.\n");
    return 1;
  }
  
  printf("Aguarde arquivo sendo gerado... ");
  
  for (I = 0; I <= 999999; I++) {
    snprintf(MATRICULA_TEMP, sizeof(MATRICULA_TEMP), "%s", formataMatricula(I));
    fprintf(ARQUIVO, "%.3s.%.3s-%d\n", MATRICULA_TEMP, MATRICULA_TEMP + 3, mod10(MATRICULA_TEMP));
  }
  
  fclose(ARQUIVO);
  
  printf("\n\nSUCESSO: Arquivo gerado.\n");
  
  return 0;
}
